src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -178,6 +243,11 @@ def __init__(self, config):
             )
         self.conv_layers = nn.ModuleList(conv_layers)
 
+        if config.freeze_feat_extract_train:
+            # freeze feature_extractor
+            for param in self.parameters():
+                param.requires_grad = False
```

Comment:
I would put this in a model method personally (and the config arg as a training argument, the script would then be responsible to call the method on the model).

The problem with putting it in the config is that it will be saved with the pretrained model which will then do this forever and ever (unless someone goes out of their way to remove it) whereas it seems just a training artifact to me. It's a bit as if we stored the learning rate in the model config and used it for all subsequents fine-tunings.

Reply:
Hmmm, I don't fully agree here since the default case is that the feature extraction layers are not fine-tuned anymore after pertaining, but I definitely see your point! So good for me to have a model method instead and call it from the fine-tuning script!

Reply:
> Also, we have configuration arguments like gradient_checkpointing (do you think they should not be in the config either?)

Yes, absolutely. I've discussed it with @LysandreJik and will make a PR when I have a bit of time to work on this. For the same reason I think `gradient_chekpointing` should be a model method that we activate in the training scripts when the right training arg is passed (of course in the case of gradient checkpointing, there will be a deprecation cycle and proper BC since it's been around for a while).